server/parsec/components/account.py

```diff
@@ -323,6 +358,56 @@ async def api_account_vault_key_rotation(
             case AccountVaultKeyRotation.ACCOUNT_NOT_FOUND:
                 client_ctx.account_not_found_abort()
 
+    @api
+    async def api_account_vault_item_recovery_list(
+        self,
+        client_ctx: AuthenticatedAccountClientContext,
+        req: authenticated_account_cmds.latest.vault_item_recovery_list.Req,
+    ) -> authenticated_account_cmds.latest.vault_item_recovery_list.Rep:
+        outcome = await self.vault_item_recovery_list(
+            auth_method_id=client_ctx.auth_method_id,
+        )
+        match outcome:
+            case VaultItemRecoveryList() as vaults:
+
+                def _convert_auth_methods(
+                    auth_method: VaultItemRecoveryAuthMethod,
+                ) -> authenticated_account_cmds.latest.vault_item_recovery_list.VaultItemRecoveryAuthMethod:
+                    match auth_method:
+                        case VaultItemRecoveryAuthMethodPassword():
+                            match auth_method.algorithm:
+                                case PasswordAlgorithmArgon2ID() as a:
+                                    algorithm = authenticated_account_cmds.latest.vault_item_recovery_list.PasswordAlgorithmArgon2id(
+                                        salt=a.salt,
+                                        opslimit=a.opslimit,
+                                        memlimit_kb=a.memlimit_kb,
+                                        parallelism=a.parallelism,
+                                    )
+
+                            return authenticated_account_cmds.latest.vault_item_recovery_list.VaultItemRecoveryAuthMethodPassword(
+                                created_on=auth_method.created_on,
+                                created_by_ip=auth_method.created_by_ip,
+                                created_by_user_agent=auth_method.created_by_user_agent,
+                                vault_key_access=auth_method.vault_key_access,
+                                algorithm=algorithm,
```

Comment:
Something looks odd here... if there is no match in `auth_method.algorithm`, the `algorithm` will not be set (None)?